controllers/templatesync/template_sync.go

```diff
@@ -243,50 +312,43 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 				break
 			}
 
-			allDeps[depID] = dep.Compliance
-			templateDeps[depID] = dep.Compliance
+			allDeps[depID] = string(dep.Compliance)
+			templateDeps[depID] = string(dep.Compliance)
 		}
 
 		// skip template if dependencies ask for conflicting compliances
 		if depConflictErr {
 			continue
 		}
 
-		object, gvk, err := unstructured.UnstructuredJSONScheme.Decode(policyT.ObjectDefinition.Raw, nil, nil)
-		if err != nil {
-			resultError = err
-			errMsg := fmt.Sprintf("Failed to decode policy template with err: %s", err)
-
-			r.emitTemplateError(instance, tIndex, fmt.Sprintf("[template %v]", tIndex), errMsg)
-			reqLogger.Error(resultError, "Failed to decode the policy template", "templateIndex", tIndex)
-
-			policyUserErrorsCounter.WithLabelValues(instance.Name, "", "format-error").Inc()
-
-			continue
-		}
-
 		var tName string
 		if tMetaObj, ok := object.(metav1.Object); ok {
 			tName = tMetaObj.GetName()
 		}
 
 		if tName == "" {
 			errMsg := fmt.Sprintf("Failed to get name from policy template at index %v", tIndex)
-			resultError = errors.NewBadRequest(errMsg)
+			resultError = k8serrors.NewBadRequest(errMsg)
 
-			r.emitTemplateError(instance, tIndex, fmt.Sprintf("[template %v]", tIndex), errMsg)
+			r.emitTemplateError(instance, tIndex, fmt.Sprintf("[template %v]", tIndex), isClusterScoped, errMsg)
 			reqLogger.Error(resultError, "Failed to process the policy template", "templateIndex", tIndex)
 
 			policyUserErrorsCounter.WithLabelValues(instance.Name, "", "format-error").Inc()
 
 			continue
 		}
 
+		childTemplateNs := ""
+
+		if !isClusterScoped {
```

Comment:
Could perform this check after `isClusterScoped` is confirmed (i.e. after `isClusterScoped = mapping.Scope == meta.RESTScopeRoot`)?

Reply:
Yes, then we can reuse the namespace from the dynamic client also. Good find!